packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -63,46 +69,23 @@ def __init__(self, namespace, parent_node, obj, is_module_level=False):
         # Some of the methods wont be listed in API review
         # For e.g. ABC methods if class implements all ABC methods
         self.hidden = False
+        self.node = node or astroid.extract_node(inspect.getsource(obj))
         self._inspect()
 
 
     def _inspect(self):
         logging.debug("Processing function {0}".format(self.name))
-        try:
-            code = inspect.getsource(self.obj).strip()
-        except OSError:
-            # skip functions with no source code
-            self.is_async = False
-            return
-        
-        for line in code.splitlines():
-            # skip decorators
-            if line.strip().startswith("@"):
-                continue
-            # the first non-decorator line should be the function signature
-            self.is_async = line.strip().startswith("async def")
-            self.def_key = "async def" if self.is_async else "def"
-            break
+
+        self.is_async = isinstance(self.node, astroid.AsyncFunctionDef)
+        self.def_key = "async def" if self.is_async else "def"
 
         # Update namespace ID to reflect async status. Otherwise ID will conflict between sync and async methods
         if self.is_async:
             self.namespace_id += ":async"
-
-        # Find decorators and any annotations
-        try:
-            node = astroid.extract_node(inspect.getsource(self.obj))
-            if node.decorators:
-                self.annotations = [
-                    "@{}".format(x.name)
-                    for x in node.decorators.nodes
-                    if hasattr(x, "name")
-                ]
-        except:
-            # TODO: Update exception details in error
-            error_message = "Error in parsing decorators for function {}".format(
-                self.name
-            )
-            self.add_error(error_message)
+        
+        # Turn any decorators into annotation
+        if self.node.decorators:
+            self.annotations = [f"@{x.as_string()}" for x in self.node.decorators.nodes]
```

Comment:
@xiangyan99 this is the only part that is relevant to the fix for your issue :)  Save the you the trouble of reviewing all the other stuff.

Reply:
Looks like yes. :)

Reply:
And here's the APIView showing the result of the fix:
https://apiviewstaging.azurewebsites.net/Assemblies/Review/94a5081dbc834524b4e680286e922dde